### DISCUSSION TITLE
Please allow returning only shows as part of the sync/history endpoint

### DISCUSSION THREAD

**Discussion by thesobercoder** [trakt/trakt-api#465] [Q&A] [answered]
It would be great if you could add the option to return only shows from the `sync/history` endpoint. In my Raycast extension, I'm looking for a way to show users a list of shows they have in their history with pagination. Currently, the endpoint returns all episodes, which includes the show information. However, if I try to extract the unique shows, the pagination doesn't represent it correctly and everything gets messed up. It would be incredibly helpful to have only the shows returned.

Something like below -

https://api.trakt.tv/sync/history/shows?extended=noepisodes

This is my Raycast extension that I'm building.

https://github.com/thesobercoder/trakt-manager

**Comment by rectifyer** [comment, marked as answer]
You can use `/sync/watched/shows?extended=noseasons` then just sort the shows in app using the `last_watched_at` field.

https://trakt.docs.apiary.io/#reference/sync/get-watched/get-watched

**Reply by thesobercoder** [reply]
Yeah, I took a look at this endpoint but sadly, it doesn't support pagination, which is what I was looking for. On a different note, I'd love for you to try the Trakt Manager and let me know your thoughts, if you happen to be on a Mac and use Raycast.

https://www.raycast.com/thesobercoder/trakt-manager
